### DISCUSSION TITLE
HoloLens 2 question - Enterprise AAD/Entra authentication

### DISCUSSION THREAD

**Discussion by laultman** [StereoKit/StereoKit#788] [Q&A]
I'm integrating Teams into an app. I have is working fine on PC in UWP (still trying to figure out 1:1 user VoIP access) but I am having trouble with the AAD authentication. What I really need is Application authentication to use Teams, but you still have to have a valid user identity (license) to initiate the connection. Like I said, it is working in PC/UWP. The reason it works in PC seems to be the fact that in the PC it can interactively acquire the access token on first contact and silently afterwards. The HoloLens 2 seems to need to acquire the access token silently from the OS (Windows Hello). I found this [article](https://mtirion.medium.com/authenticate-a-hololens-application-to-the-azure-ad-ad551fbb8440) but I'm getting a memory access violation in corelib.

Has anyone got this working anywhere? I'm going to post this also on stackoverflow.

**Comment by laultman** [comment]
It's not easy. See WAM.

**Reply by maluoi** [reply]
There's an example [up over here by Nakamir](https://github.com/Nakamir-Code/SKAzureCloud)! It might be interesting reference, but they aren't making any promises about it actually working still :) I guess this stuff is just complicated and not fun to maintain.
